integration/benchmark/main.go

```diff
@@ -0,0 +1,372 @@
+// remain in sync with tps_results_schema.json
+type dataSlice struct {
+	GitSha              string
+	StartTime           time.Time
+	EndTime             time.Time
+	InputTps            float64
+	OutputTps           float64
+	ProStartTransaction float64
+	ProEndTransaction   float64
+}
+
+func main() {
+	// holdover flags from loader/main.go
+	sleep := flag.Duration("sleep", 0, "duration to sleep before benchmarking starts")
+	loadTypeFlag := flag.String("load-type", "token-transfer", "type of loads (\"token-transfer\", \"add-keys\", \"computation-heavy\", \"event-heavy\", \"ledger-heavy\", \"const-exec\")")
+	chainIDStr := flag.String("chain", string(flowsdk.Emulator), "chain ID")
+	access := flag.String("access", net.JoinHostPort("127.0.0.1", "3569"), "access node address")
+	serviceAccountPrivateKeyHex := flag.String("servPrivHex", unittest.ServiceAccountPrivateKeyHex, "service account private key hex")
+	logLvl := flag.String("log-level", "info", "set log level")
+	metricport := flag.Uint("metricport", 8080, "port for /metrics endpoint")
+	pushgateway := flag.String("pushgateway", "127.0.0.1:9091", "host:port for pushgateway")
+	profilerEnabled := flag.Bool("profiler-enabled", false, "whether to enable the auto-profiler")
+	accountMultiplierFlag := flag.Int("account-multiplier", 50, "number of accounts to create per load tps")
+	feedbackEnabled := flag.Bool("feedback-enabled", true, "wait for trannsaction execution before submitting new transaction")
+	maxConstExecTxSizeInBytes := flag.Uint("const-exec-max-tx-size", flow.DefaultMaxTransactionByteSize/10, "max byte size of constant exec transaction size to generate")
+	authAccNumInConstExecTx := flag.Uint("const-exec-num-authorizer", 1, "num of authorizer for each constant exec transaction to generate")
+	argSizeInByteInConstExecTx := flag.Uint("const-exec-arg-size", 100, "byte size of tx argument for each constant exec transaction to generate")
+	payerKeyCountInConstExecTx := flag.Uint("const-exec-payer-key-count", 2, "num of payer keys for each constant exec transaction to generate")
+
+	// CI relevant flags
+	tpsFlag := flag.String("tps", "300", "transactions per second (TPS) to send, accepts a comma separated list of values if used in conjunction with `tps-durations`")
+	tpsDurationsFlag := flag.String("tps-durations", "10m", "duration that each load test will run, accepts a comma separted list that will be applied to multiple values of the `tps` flag (defaults to infinite if not provided, meaning only the first tps case will be tested; additional values will be ignored)")
+	ciFlag := flag.Bool("ci-run", false, "whether or not the run is part of CI")
+	leadTime := flag.String("leadTime", "30s", "the amount of time before data slices are started")
+	sliceSize := flag.String("sliceSize", "2m", "the amount of time that each slice covers")
+	gitSha := flag.String("gitSha", "", "the git hash of the run, used for BigQuery result tracking")
```

Comment:
commit and semver are injected during the build, so you can use them through cmd/build/version.go

Reply:
gitSha, goVersion and osVersion are now pulled directly from version and runtime.